lib/services/protocol/service.js

```diff
@@ -70,17 +70,73 @@ function processSystemOut(message, callback) {
 
 function processInboundStack(message, protocol, callback) {
   var transformed;
-
+  let _this = this;
   try {
     transformed = protocol.transformIn(message);
     if (transformed.__suppress) return callback(null, message);
   } catch (e) {
     return callback(e);
   }
 
+  if (!this.config.allowNestedPermissions) return this.__processSinglePath(transformed, callback);
+
+  if (
+    transformed.request &&
+    transformed.request.path &&
+    transformed.request.path.indexOf('**') > -1 &&
+    (transformed.request.action === 'on' || transformed.request.action === 'get')
+  ) {
+    this.happn.services.security.getRelevantPaths(transformed, (e, relevantPaths) => {
+      if (e) return callback(e);
+      if (relevantPaths.allowed.length === 0) {
+        return this.__processSinglePath(transformed, callback);
+      }
+
+      if (transformed.request.action === 'on') {
+        transformed.request = { ...transformed.request, ...relevantPaths };
+        return this.happn.services.subscription.processMultiSubscribe(
+          this.happn.services.subscription.prepareSubscribeMessage(transformed),
+          function(e, result) {
+            if (e) return callback(e);
+            callback(null, result);
+          }
+        );
+      }
+
+      // (If action == get)
+      let clonedMsg = JSON.parse(JSON.stringify(transformed));
```

Comment:
Why we cloning here? Unless there is a good reason we must avoid cloning...

Reply:
I think this can be taken out

Reply:
It gone.